libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt

```diff
@@ -699,6 +705,65 @@ internal class MapRouteLine(
         }?.lineString ?: LineString.fromPolyline(route.geometry()!!, Constants.PRECISION_6)
     }
 
+    /**
+     * The map will be queried for a route line feature at the target point or a bounding box
+     * centered at the target point with a padding value determining the box's size. If a route
+     * feature is found the index of that route in this class's route collection is returned. The
+     * primary route is given precedence if more than one route is found.
+     *
+     * @param target a target latitude/longitude serving as the search point
+     * @param mapboxMap a reference to the MapboxMap that will be queried
+     * @param padding a sizing value added to all sides of the target point  for creating a bounding
+     * box to search in.
+     *
+     * @return the index of the route in this class's route collection or -1 if no routes found.
+     */
+    fun findClosestRoute(
+        target: LatLng,
+        mapboxMap: MapboxMap,
+        padding: Int
+    ): Int {
+        val mapClickPointF = mapboxMap.projection.toScreenLocation(target)
+        val leftFloat = (mapClickPointF.x - padding)
+        val rightFloat = (mapClickPointF.x + padding)
+        val topFloat = (mapClickPointF.y - padding)
+        val bottomFloat = (mapClickPointF.y + padding)
+        val clickRectF = RectF(leftFloat, topFloat, rightFloat, bottomFloat)
+
+        val primaryFeaturesOnPoint = mapboxMap.queryRenderedFeatures(
```

Comment:
`queryRenderedFeatures` calls are relatively expensive. Let's execute the call and evaluate the result immediately before falling back to another one rather than executing all at once.